[
  {
    "url": "https://api.github.com/repos/ipython/ipython/issues/comments/901423",
    "html_url": "https://github.com/ipython/ipython/issues/214#issuecomment-901423",
    "issue_url": "https://api.github.com/repos/ipython/ipython/issues/214",
    "id": 901423,
    "node_id": "MDEyOklzc3VlQ29tbWVudDkwMTQyMw==",
    "user": {
      "login": "takluyver",
      "id": 327925,
      "node_id": "MDQ6VXNlcjMyNzkyNQ==",
      "avatar_url": "https://avatars3.githubusercontent.com/u/327925?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/takluyver",
      "html_url": "https://github.com/takluyver",
      "followers_url": "https://api.github.com/users/takluyver/followers",
      "following_url": "https://api.github.com/users/takluyver/following{/other_user}",
      "gists_url": "https://api.github.com/users/takluyver/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/takluyver/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/takluyver/subscriptions",
      "organizations_url": "https://api.github.com/users/takluyver/orgs",
      "repos_url": "https://api.github.com/users/takluyver/repos",
      "events_url": "https://api.github.com/users/takluyver/events{/privacy}",
      "received_events_url": "https://api.github.com/users/takluyver/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2011-03-22T03:06:14Z",
    "updated_at": "2011-03-22T03:06:14Z",
    "author_association": "MEMBER",
    "body": "Within Python, there's a getpass module for this sort of thing. But we have to send user input on to a running process, and I don't think there's a good general way to discover when the process wouldn't echo stdin. As a simple workaround, we could make a sudo magic function which would use getpass, then run the process and silently feed it that password.\n"
  },
  {
    "url": "https://api.github.com/repos/ipython/ipython/issues/comments/7297149",
    "html_url": "https://github.com/ipython/ipython/issues/214#issuecomment-7297149",
    "issue_url": "https://api.github.com/repos/ipython/ipython/issues/214",
    "id": 7297149,
    "node_id": "MDEyOklzc3VlQ29tbWVudDcyOTcxNDk=",
    "user": {
      "login": "bfroehle",
      "id": 204333,
      "node_id": "MDQ6VXNlcjIwNDMzMw==",
      "avatar_url": "https://avatars3.githubusercontent.com/u/204333?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/bfroehle",
      "html_url": "https://github.com/bfroehle",
      "followers_url": "https://api.github.com/users/bfroehle/followers",
      "following_url": "https://api.github.com/users/bfroehle/following{/other_user}",
      "gists_url": "https://api.github.com/users/bfroehle/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/bfroehle/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/bfroehle/subscriptions",
      "organizations_url": "https://api.github.com/users/bfroehle/orgs",
      "repos_url": "https://api.github.com/users/bfroehle/repos",
      "events_url": "https://api.github.com/users/bfroehle/events{/privacy}",
      "received_events_url": "https://api.github.com/users/bfroehle/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2012-07-27T04:07:42Z",
    "updated_at": "2012-07-27T04:07:51Z",
    "author_association": "CONTRIBUTOR",
    "body": "Two points:\n1. I cannot reproduce this on Ubuntu 12.04.\n2. There are several potential workarounds, including:\n- Hit ctrl-z to drop back to your shell and run sudo there.\n- Run !bash to get a shell prompt, then run sudo.\n\nFor these reasons, I'm going to mark the issue as closed.\n"
  }
]
